src/php/ext/grpc/channel.c

```diff
@@ -222,10 +361,42 @@ PHP_METHOD(Channel, watchConnectivityState) {
  */
 PHP_METHOD(Channel, close) {
   wrapped_grpc_channel *channel = Z_WRAPPED_GRPC_CHANNEL_P(getThis());
-  if (channel->wrapped != NULL) {
-    grpc_channel_destroy(channel->wrapped);
-    channel->wrapped = NULL;
+  if (channel->wrapper->wrapped != NULL) {
+    gpr_mu_lock(&channel->wrapper->mu);
```

Comment:
I think you need to wrap this whole ```if``` block with the mutex, and also aquire the mutex wherever you have a ```if(chanel->wrapper->wrapped != NULL)``` block.